llvm/lib/LTO/LTO.cpp

```diff
@@ -91,6 +94,19 @@ extern cl::opt<bool> SupportsHotColdNew;
 
 /// Enable MemProf context disambiguation for thin link.
 extern cl::opt<bool> EnableMemProfContextDisambiguation;
+
+cl::list<std::string> AdditionalThinLTODistributorArgs(
+    "thinlto-distributor-arg",
+    cl::desc("Additional arguments to pass to the ThinLTO distributor"));
+
+cl::opt<std::string> ThinLTORemoteCompiler(
```

Comment:
Are those `cl::opt` debugging option or the real option that user needs to pass in? I should avoid `-mllvm` option in production usage.

Reply:
> Are those `cl::opt` debugging option or the real option that user needs to pass in? I should avoid `-mllvm` option in production usage.

The use of -mllvm options avoids the need to implement a similar set of options for DTLTO in each LLD port (and also in llvm-lto2). See: https://github.com/llvm/llvm-project/pull/126654#discussion_r1966932717. I don't have a strong opinion though. Perhaps @maskray would like to comment on this.

Reply:
I'm happy to remove the cl::opt options and instead pass extra state when configuring the new backend. I will implement that in this PR.

Reply:
I understand that -mllvm options are considered internal and for production usage we should recommend proper driver and linker options. It's annoying though to update llvm-lto2 and every ELF port....

Reply:
Using -mllvm options was a nice reduction in the number of changes required in different tools, which was helpful to reduce the size of the initial PR. However, I think there is agreement now that proper options are required. Note that we don't have to update all the LLD ports at once. We can add DTLTO support to the different LLD ports incrementally. Perhaps we can start with just the ELF LLD port (chosen as it is the LLD port I'm most familiar with)?